engine/classes/ElggEntity.php

```diff
@@ -1670,6 +1697,8 @@ protected function update() {
 
 		_elgg_cache_entity($this);
 
+		$this->orig_attributes = [];
```

Comment:
Actually, this should be good enough. Plugin code doesn't have access to the entity between the event and the complete saving of the entity.